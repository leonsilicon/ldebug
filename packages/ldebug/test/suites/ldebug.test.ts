import { expect, test } from 'vitest';
import { spyOnImplementing } from 'vitest-mock-process';

import { createDebug } from '~/index.js';

test('works', () => {
	const debug = createDebug({
		isDevelopment: true,
	});

	const mockDebug = spyOnImplementing(console, 'debug', () => {
		/* noop */
	});

	debug((f) => f`hello ${['world']}`);

	expect(mockDebug.mock.calls).toMatchSnapshot();
});

test('works without prettifying', () => {
	const debug = createDebug({
		isDevelopment: true,
		prettyStringify: false,
		highlight: false,
	});

	const mockDebug = spyOnImplementing(console, 'debug', () => {
		/* noop */
	});

	debug((f) => f`hello ${['world']}`);

	expect(mockDebug).toBeCalledWith('hello ["world"]');
});
