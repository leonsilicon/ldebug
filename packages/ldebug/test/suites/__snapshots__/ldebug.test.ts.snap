// Vitest Snapshot v1

exports[`works 1`] = `
[
  [
    "hello [[31m\\"world\\"[39m]",
  ],
]
`;
